### DISCUSSION TITLE
Localisations for keybinds haven't been pulled for some languages

### DISCUSSION THREAD

**Discussion by Walavouchey** [ppy/osu#15932] [Q&A] [answered]
There are Finnish localisations for all of the keybinds (by yours truly), as well as almost everything in Ukrainian, but they haven't been added to the game yet as it seems. Other languages with full localisations have been put in.

I was waiting on posting about this since I thought they'd be coming in a few localisation batches, but now I'm guessing there may be another reason.

https://user-images.githubusercontent.com/36758269/144715461-cc91dffa-fd1a-4b8f-bb0f-189da4f9e4a6.mp4

There are other partial translations available on Crowdin that haven't been pulled:

- Czech
- Polish
- Portuguese
- Portuguese, Brazilian
- Swedish

**Comment by bdach** [comment]
~~the components in question aren't handling localisations correctly yet. this'll be resolved when that support is added.~~

edit 1: nvm, i thought this was not supported yet. looks to be an actual issue, investigating further.

edit 2: on checking locally, this works here. i can't seem to reproduce.

**Reply by Walavouchey** [reply]
can now reproduce from local git repo on debug and release mode (it magically fixed itself), but the latest release doesn't work.

tried:
- deleting the `AppData\Local\osulazer\app-2021.1202.0` folder and running the latest installer in hopes of redownloading stale dll files, but no dice (newest dll files didn't work)
- downloading the source code from the release page and building local release and debug builds - no dice
- replacing my lazer install's dll files (`fi\osu.Game.Resources.resources.dll` and `fi-FI\Humanizer.resources.dll`) in `%APPDATALOCAL%\osulazer\app-2021.1202.0` with the ones from my local git repo (`osu.Desktop\bin\Release\net5.0`) - **WORKS**

i have concluded that these dll files (as tested for the finnish locale) in the release builds are outdated:
[finnish-lazer-localisation-RELEASE.zip](https://github.com/ppy/osu/files/7654328/finnish-lazer-localisation-RELEASE.zip) (github release [2021.1202.0](https://github.com/ppy/osu/releases/tag/2021.1202.0))

works with below locally built files:
[finnish-lazer-localisation-LOCAL_RELEASE.zip](https://github.com/ppy/osu/files/7654332/finnish-lazer-localisation-LOCAL_RELEASE.zip) (commit 70e818ff8f5f5e54aca55752b37d71e46115de73 on master)

**Reply by bdach** [reply]
on triple-checking the strings just weren't there last release, and resources haven't been bumped since then. next release should contain the finnish strings, if it doesn't then we'll revisit.

**Comment by Walavouchey** [comment, marked as answer]
fixed in https://github.com/ppy/osu/commit/614256697442c2c543d0120d1271be1264dfb858, working build at https://github.com/ppy/osu/commit/9803e63e6f0c9a2a14f6427c9faafb629c5ef2ac (next commit)
